### PR TITLE
Fix error handling in integration test scaffolding

### PR DESCRIPTION
**Pull request by azdagron**
The integration test scaffolding has fallen prey to bash-ism whereby `set -e` is NOT honored in subshells, even if set explicitly. This has caused integration tests to continue running when a command in a step script fails when the exit status not explicitly tested, hiding test failures.

The fix is to NOT execute the test in a subshell, but rather invoke a separate instance of bash against a copy of the step code with a preamble containing the common functions which also sets up bash options like errexit and pipefail.

The change also includes a fix to the nested rotation test which had a failed invocation due to a missing docker-compose option disabling the TTY.

There is also included a small self-test suite that ensures that command failure fails step scripts. More tests can be added in the future that ensure various mechanics of the integration test framework are working as expected.

### CODE REVIEW

File: test/integration/test-one.sh [hunk 1]
```diff
@@ -17,27 +17,28 @@ REPODIR=$(git rev-parse --show-toplevel)
 
 # Set and export the PATH to one that includes a go binary installed by the
 # Makefile, if necessary.
-PATH=$(cd "${REPODIR}"; make go-bin-path)
+PATH=$(cd "${REPODIR}" || exit; make go-bin-path)
```
Comment (evan2645) [inline test/integration/test-one.sh:20]: Will we get to see the error logged by `cd` here if we end up having to invoke `exit` ?

**Review by evan2645 — APPROVED** [review]
Thanks for patching this @azdagron

+1 creativity point for the self-test  :)

File: test/integration/test-one.sh (hunk 1 above)
Comment (azdagron) [reply test/integration/test-one.sh:20]: Yup. `cd` will emit the failure (e.g. `bash: cd: whatever: No such file or directory`) to stderr.

File: test/integration/suites/nested-rotation/10-check-svids [hunk 2]
```diff
@@ -18,7 +18,7 @@ validateX509SVID() {
       -socketPath /opt/spire/sockets/workload_api.sock \
       -write /tmp || fail-now "x509-SVID check failed"
 
-  docker-compose exec $2 openssl verify -verbose -CAfile /tmp/bundle.0.pem -untrusted /opt/svid.0.pem /opt/svid.0.pem
+  docker-compose exec -T $2 openssl verify -verbose -CAfile /tmp/bundle.0.pem -untrusted /opt/svid.0.pem /opt/svid.0.pem
```
[the remark this answers is not in the document]
Comment (azdagron) [reply test/integration/suites/nested-rotation/10-check-svids:21]: :+1:

**Review by evan2645 — APPROVED** [review]

**Pull request merged**